src/engine/Engine.cpp

```diff
@@ -1924,6 +1925,15 @@ void Engine::checkOverallProgress()
     }
 }
 
+void Engine::updateDirections()
+{
+    if ( GlobalConfiguration::USE_POLARITY_BASED_DIRECTION_HEURISTICS )
+        for ( const auto &constraint : _plConstraints )
+            if ( constraint->supportPolarity() &&
+                 constraint->isActive() && !constraint->phaseFixed() )
+                ( (ReluConstraint *) constraint )->updateDirection();
```

Comment:
here you are making the assumption that if the plconstraint supports polarity then it must be a reluconstraint. right?

Reply:
Added updateDirection as a virtual method in PiecewiseLinearConstraint.h that does nothing by default.